app/models/allocation.rb

```diff
@@ -19,13 +22,23 @@
 class Allocation < ApplicationRecord
   belongs_to :provider
   belongs_to :accredited_body, class_name: "Provider"
+  belongs_to :recruitment_cycle
 
   validates :provider, :accredited_body, presence: true
-  validate :accredited_body_is_an_accredited_body
   validates :number_of_places, numericality: true
 
+  validate :accredited_body_is_an_accredited_body
+
   enum request_type: { initial: 0, repeat: 1, declined: 2 }
 
+  def previous
+    Allocation.find_by(
+      provider_code: provider_code,
```

Comment:
Can we add indexes for these three columns?

Reply:
done